continuous-integration/requirements.Rmd

```diff
@@ -0,0 +1,28 @@
+---
+title: "Requirements for continuous integration"
+maintainer: "Thierry Onkelinx"
+output: html_document
+---
+
+# Must have
+
+- start automatically when pushing a git repository on Github or Bitbucket
+- start automatically when creating a pull request on Github or Bitbucket
+- able to use a Docker image to setup the environment in which CI runs
+- able to set environment variables
+- able to check code style with lintr and report the results back to the SCM
+- able to check code coverage with covr and report the results back to the SCM
+- a succesful build creates a Docker image
+- a succesful build on specific branches can be pushed to Docker hub (e.g. https://hub.docker.com/r/inbonatura2000/n2khelper/tags/)
+- configuration of CI can be partially through a configuration file and partially through a webapplication
```

Comment:
As such, I do not completely agree on the this requirement. I do absolutely not like the idea that configuration should be partially done by config-file and partially by a web-application. I think the requirement should be that it can be **fully configured with a configuration file which is completely self-containing**. The latter is a requirement, and a UI is more a nice-to-have (if and only if the things you can do are also captured by the config-file).